rpc_util.go

```diff
@@ -261,28 +261,27 @@ func encode(c Codec, msg interface{}, cp Compressor, cbuf *bytes.Buffer, outPayl
 	var (
 		b      []byte
 		length uint
+		err    error
```

Comment:
all the changes in this file are unnecessary (and the new code is worse, since it increases the scope of `err` needlessly).